test/test_braid.py

```diff
@@ -569,14 +570,18 @@ def test_performable_moves(self, sigmas):
             states.append(move())
 
         all_moves = self.get_all_moves(braid)
-        
+
+        all_states = []       
         for move in all_moves:
             try:
                 state = move()
             except IllegalTransformationException as e:
                 pass
             else:
-                assert state in states
+                all_states.append(state)
+        assert len(states) == len(all_states)
+        for state in states:
+            assert state in all_states
```

Comment:
This fix should've been applied to `test_performable_moves_random` as well. Maybe remove code duplication so this type of mistake can't be made in the future?